csharp/src/Ice/Protocol.cs

```diff
@@ -2,10 +2,162 @@
 // Copyright (c) ZeroC, Inc. All rights reserved.
 //
 
+using System;
+using System.Diagnostics;
+
 namespace IceInternal
 {
+     // TODO: move class to namespace Ice
     public static class Protocol
     {
+        /// <summary>Starts a new response frame and the encapsulation in that frame. This frame encodes a "success"
+        /// response.</summary>
+        /// <param name="current">The current parameter holds decoded header data and other information about the
+        /// request currently dispatched.</param>
+        /// <param name="format">The Slice format (Compact or Sliced) used by the encapsulation.</param>
+        /// <returns>An <see cref="Ice.OutputStream"/> that holds the new frame.</returns>
+        public static Ice.OutputStream StartResponseFrame(Ice.Current current, Ice.FormatType? format = null)
+        {
+            Debug.Assert(format == current.Format); // temporary
+
+            var ostr = new Ice.OutputStream(current.Adapter.Communicator, Ice.Util.CurrentProtocolEncoding);
+            ostr.WriteBlob(Protocol.replyHdr);
+            ostr.WriteInt(current.RequestId);
+            ostr.WriteByte(ReplyStatus.replyOK);
+            ostr.StartEncapsulation(current.Encoding, format);
+            return ostr;
+        }
+
+        /// <summary>Creates a new response frame that represents "success" with a void return value. The returned
+        /// frame is complete and no additional data can be included in its payload.</summary>
+        /// <param name="current">The current parameter holds decoded header data and other information about the
+        /// request currently dispatched.</param>
+        /// <returns>An <see cref="Ice.OutputStream"/> that holds the new frame.</returns>
+        public static Ice.OutputStream CreateEmptyResponseFrame(Ice.Current current)
```

Comment:
We could also name it EmptyResponseFrame or VoidResponseFrame? Same for CreateResponseFrameForFailure, we could name it FailureResponseFrame.

Reply:
Since these are factory methods, a prefix like Create makes sense. What about `CreateEmptyResponseFrame` and `CreateFailureResponseFrame`?

Reply:
Fine with me.

Reply:
Fixed.